### PR TITLE
Support JIT invoicing on donations to rewards pool

### PR DESCRIPTION
**Pull request by SatsAllDay**
Now you can just-in-time fund your account to donate to SN's reward pool.

You can also donate without an account via the `anon` account, also using JIT invoices.

### CODE REVIEW

File: api/resolvers/item.js [hunk 1]
```diff
@@ -38,7 +38,10 @@ export async function commentFilterClause (me, models) {
   return clause
 }
 
-async function checkInvoice (models, hash, hmac, fee) {
+export async function checkInvoice (models, hash, hmac, fee) {
```
Comment (SatsAllDay) [inline api/resolvers/item.js:41]: Exported for use in rewards resolver

File: api/resolvers/item.js [hunk 2]
```diff
@@ -38,7 +38,10 @@ export async function commentFilterClause (me, models) {
...
+  if (!hash) {
+    throw new GraphQLError('hash required', { extensions: { code: 'BAD_INPUT' } })
+  }
```
Comment (SatsAllDay) [inline api/resolvers/item.js:44]: Probably overkill but just for the sake of explicit checking

File: api/resolvers/rewards.js [hunk 3]
```diff
@@ -102,17 +103,38 @@ export default {
...
+      }
+      let user
+      if (me) {
+        user = me
+      }
+
+      let invoice
+      if (hash) {
+        invoice = await checkInvoice(models, hash, hmac, sats)
+        user = invoice.user
```
Comment (SatsAllDay) [inline api/resolvers/rewards.js:118]: This will cover the anon case

File: pages/rewards/index.js [hunk 4]
```diff
@@ -114,6 +117,7 @@ export function DonateButton () {
           <Input
             label='amount'
             name='amount'
+            type='number'
```
Comment (SatsAllDay) [inline pages/rewards/index.js:120]: This wasn't an issue before (meaning you could type other numbers beyond 1000), but the invoicing code was failing due to type validation. Of course, it is an issue allowing alpha chars haha

**Comment by SatsAllDay** [comment]
Really great to have all the reusable code to make this happen quickly and reliably!

**Comment by huumn** [comment]
Amaze! I'll review this tomorrow morning but it looks good at first glance

**Pull request merged**